rust_src/src/buffers.rs

```diff
@@ -320,6 +335,57 @@ pub fn overlay_buffer(overlay: LispObject) -> LispObject {
     marker_buffer(marker)
 }
 
+pub fn recenter_overlay_lists(mut buf: LispBufferRef, pos: ptrdiff_t) {
+    // let mut overlay: LispObject;
+    let (mut prev, mut tail, mut next): (
+        Option<LispOverlayRef>,
+        LispOverlayRef,
+        Option<LispOverlayRef>,
+    ) = (
+        None,
+        LispOverlayRef::new((*buf).overlays_before as *mut Lisp_Overlay),
+        None,
+    );
+    loop {
+        if tail.as_ptr() as ptrdiff_t == 0 {
+            break;
+        }
+        let (beg, end, curNext) = (tail.start(), tail.end(), tail.next);
+        if end.as_fixnum_coerce_marker_or_error() as ptrdiff_t > pos {
```

Comment:
This is almost a translation from the [original](https://github.com/Wilfred/remacs/blob/aabf185e251f8586e1e4003af7fc388dcbf537e8/src/buffer.c#L3327), but it doesn't seem right to implement an Iterator here, due to performance concerns. Does that make sense?

Reply:
I say get it working then consider an iterator. If this code is only used in one place I would not worry about the iterator. Unless of course the iterator makes the code easier/better/faster.